src/fermilib/utils/_jellium_trotter_error.py

```diff
@@ -0,0 +1,461 @@
+    term_op_c, = term_c.terms.keys()
+
+    modes_touched_c = [term_op_c[0][0], term_op_c[1][0]]
+
+    # If there's no intersection between the modes term_b and term_c act
+    # on, the commutator is trivially zero.
+    if not (term_op_b[0][0] in modes_touched_c or
+            term_op_b[1][0] in modes_touched_c):
+        return True
+
+    # In the jellium Hamiltonian, possible number operators take the
+    # form a^ a or a^ b^ a b. Test for this.
+    term_b_is_number_operator = (term_op_b[0][0] == term_op_b[1][0] or
+                                 term_op_b[1][1])
+    term_c_is_number_operator = (term_op_c[0][0] == term_op_c[1][0] or
+                                 term_op_c[1][1])
+
+    # Number operators always commute.
+    if term_b_is_number_operator and term_c_is_number_operator:
+        return True
+
+    # If the first commutator's terms are both hopping, and both create
+    # or annihilate the same mode, then the result is zero.
+    if not (term_b_is_number_operator or term_c_is_number_operator):
+        if (term_op_b[0][0] == term_op_c[0][0] or
+                term_op_c[1][0] == term_op_b[1][0]):
+            return True
+
+    # The modes term_a acts on are only needed if we reach this stage.
+    term_op_a, = term_a.terms.keys()
+    modes_touched_b = [term_op_b[0][0], term_op_b[1][0]]
+    modes_touched_bc = [term_op_b[0][0], term_op_b[1][0],
+                        term_op_c[0][0], term_op_c[1][0]]
+
+    # If the term_a shares no indices with bc, the double commutator is zero.
+    if not (term_op_a[0][0] in modes_touched_bc or
+            term_op_a[1][0] in modes_touched_bc):
+        return True
+
+    # If term_b and term_c are not both number operators and act on the
+    # same modes, the commutator is zero.
+    if (sum(1 for i in modes_touched_b if i in modes_touched_c) > 1 and
+            (term_b_is_number_operator or term_c_is_number_operator)):
+        return True
+
+    # Create a list of all the creation and annihilations performed.
+    all_changes = term_op_a + term_op_b + term_op_c
+    counts = {}
+    for operator in all_changes:
+        counts[operator[0]] = counts.get(operator[0], 0) + 2 * operator[1] - 1
+
+    # If the final result creates or destroys the same mode twice.
+    commutes = max(itervalues(counts)) > 1 or min(itervalues(counts)) < -1
+
+    return commutes
+
+
+def jellium_error_operator(terms, indices=None, is_hopping_operator=None,
+                           series_order=2):
+    """Determine the difference between the exact generator of unitary
+    evolution and the approximate generator given by Trotter-Suzuki
+    to the given order.
+
+    Args:
+        terms: a list of FermionOperators in the Hamiltonian
+            to be simulated.
+        indices: a set of indices the terms act on in the same order as terms.
+        is_hopping_operator: a list of whether each term is a hopping operator.
+        series_order: the order at which to compute the BCH expansion.
```

Comment:
I would prefer if "series_order" is just not a parameter and it is clearly explained that this is for the second order formula. Because I never intend to implement other orders and I doubt others will. So its just confusing to have a parameter which must take a certain value or the function raises an exception.